### PR TITLE
Updated leak detector to separate native/java, threaded conn test

### PR DESCRIPTION
**Pull request by justinboswell**
Sample output:
```
> mvn test -Dtest=HttpClientConnectionManagerTest -DargLine="-Daws.crt.memory.tracing=true"
```
```
...
[ERROR] testMaxParallelRequests(software.amazon.awssdk.crt.test.HttpClientConnectionManagerTest)  Time elapsed: 6.025 s  <<< FAILURE!
java.lang.AssertionError: 
Potential Java Memory Leak!

JVM Usage Deltas: [3392]
JVM Samples: [1236856, 1240248]
Native Usage Deltas: [0]
Native Samples: [5120, 5120]

        at software.amazon.awssdk.crt.test.HttpClientConnectionManagerTest.testMaxParallelRequests(HttpClientConnectionManagerTest.java:181)

[ERROR] testParallelRequests(software.amazon.awssdk.crt.test.HttpClientConnectionManagerTest)  Time elapsed: 5.142 s  <<< FAILURE!
java.lang.AssertionError: 
Potential Java Memory Leak!

JVM Usage Deltas: [1984]
JVM Samples: [1251464, 1253448]
Native Usage Deltas: [0]
Native Samples: [5120, 5120]

        at software.amazon.awssdk.crt.test.HttpClientConnectionManagerTest.testParallelRequests(HttpClientConnectionManagerTest.java:169)
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/software/amazon/awssdk/crt/CRT.java [hunk 1]
```diff
@@ -154,7 +158,7 @@ private static void loadLibraryFromJar() {
     }
 
     // Called internally when bootstrapping the CRT, allows native code to do any static initialization it needs
-    private static native void awsCrtInit(boolean memoryTracingEnabled) throws CrtRuntimeException;
+    private static native void awsCrtInit(int  memoryTracingLevel) throws CrtRuntimeException;
```
Comment (ColdenCullen) [inline src/main/java/software/amazon/awssdk/crt/CRT.java]: double spaced

File: src/native/http_request_response.c [hunk 2]
```diff
@@ -254,11 +254,12 @@ void s_cache_crt_http_stream_handler(JNIEnv *env) {
 }
 
 static jobjectArray s_java_headers_array_from_native(
+    JNIEnv *env,
     struct http_stream_callback_data *callback,
     const struct aws_http_header *header_array,
     size_t num_headers) {
 
-    JNIEnv *env = aws_jni_get_thread_env(callback->jvm);
+    // JNIEnv *env = aws_jni_get_thread_env(callback->jvm);
```
Comment (ColdenCullen) [inline src/native/http_request_response.c]: del

File: src/native/http_request_response.c [hunk 3]
```diff
@@ -569,12 +570,24 @@ static bool s_fill_out_request(
         jbyteArray jname = (*env)->GetObjectField(env, jHeader, s_http_header.name);
         jbyteArray jvalue = (*env)->GetObjectField(env, jHeader, s_http_header.value);
 
+        size_t name_len = (*env)->GetArrayLength(env, jname);
+        jbyte *name = (*env)->GetPrimitiveArrayCritical(env, jname, NULL);
+        struct aws_byte_cursor name_cursor = aws_byte_cursor_from_array(name, name_len);
+
+        size_t value_len = (*env)->GetArrayLength(env, jvalue);
+        jbyte *value = (*env)->GetPrimitiveArrayCritical(env, jvalue, NULL);
```
Comment (ColdenCullen) [inline src/native/http_request_response.c]: the lens and jbyte* can all be const

File: src/native/crt.c [hunk 4]
```diff
@@ -90,22 +127,25 @@ static void s_alloc_tracker_track(struct alloc_tracker *tracker, void *ptr, size
...
-    int stack_depth = backtrace(stack_frames, AWS_ARRAY_SIZE(stack_frames));
-    char **symbols = backtrace_symbols(stack_frames, stack_depth);
-    struct aws_byte_cursor newline = aws_byte_cursor_from_c_str("\n");
-    for (int idx = 2; idx < stack_depth && idx < 10; ++idx) {
-        if (idx > 2) {
-            aws_byte_buf_append(&alloc->stacktrace, &newline);
+#if defined(ALLOC_TRACE_AVAILABLE)
+    if (s_memory_tracing == 2) {
+        /* capture stack frames */
+        void *stack_frames[2 + ALLOC_TRACING_FRAMES];
```
Comment (ColdenCullen) [inline src/native/crt.c]: Can you pull this 2 out somewhere like a const? It's also used in that memcpy below and that spooks me

File: src/native/crt.c [hunk 5]
```diff
@@ -90,22 +127,25 @@ static void s_alloc_tracker_track(struct alloc_tracker *tracker, void *ptr, size
...
+        int stack_depth = backtrace(stack_frames, AWS_ARRAY_SIZE(stack_frames));
```
Comment (ColdenCullen) [inline src/native/crt.c]: just slam stack_depth on stacktrace_t so you don't have to count it again later

Comment (justinboswell) [reply src/native/crt.c]: I fixed it in the version I have for common, I'm not gonna bother here. Also, space savings!

File: src/native/crt.c (hunk 4 above)
Comment (justinboswell) [reply src/native/crt.c]: I'm gonna replace this with the one from common where I've cleaned all this up and made it runtime configurable and unit tested and such.

File: src/native/http_request_response.c (hunk 3 above)
Comment (justinboswell) [reply src/native/http_request_response.c]: the jbytes cant, because I have to hand them back to the JVM as mutable pointers...

**Pull request merged**